### PR TITLE
[Preprocessing] Add pad to MMA intrinsic size pass

### PR DESCRIPTION
**Pull request by kuhar**
Currently this only supports convolutions; batch matmul support can easily be restored.

This is based on the following commits from the SDXL branch:
1. https://github.com/openxla/iree/pull/16774 by @raikonenfnu 
2. https://github.com/openxla/iree/pull/16809 by @MaheshRavishankar 
3. https://github.com/openxla/iree/pull/16816 by @MaheshRavishankar 
4. https://github.com/openxla/iree/pull/16860 (batch matmul disabled) by @hanhanW

with some clean ups and tests added.

### CODE REVIEW

File: compiler/src/iree/compiler/Preprocessing/Common/PadToIntrinsics.cpp [hunk 1]
```diff
@@ -0,0 +1,220 @@
...
+    auto intrinsics = llvm::map_to_vector(
+        mmaKinds.getAsRange<IREE::GPU::MMAAttr>(), [](IREE::GPU::MMAAttr mma) {
+          auto [mSize, nSize, kSize] = mma.getMNKShape();
+          auto [aType, bType, cType] = mma.getABCElementTypes();
+          return GPUMatmulShapeType{mSize, nSize, kSize, aType, bType, cType};
+        });
+
+    SmallVector<linalg::LinalgOp> targetOps;
+    rootOp->walk([&](linalg::LinalgOp linalgOp) {
+      if (isa<linalg::Conv2DNhwcHwcfOp, linalg::BatchMatmulOp>(
```
Comment (raikonenfnu) [inline compiler/src/iree/compiler/Preprocessing/Common/PadToIntrinsics.cpp]: NIT: Maybe we can drop the BMMOp for now.

**Review by raikonenfnu — APPROVED** [review]
Thank you so much for landing this man, overall looks great!

File: compiler/src/iree/compiler/Preprocessing/Common/PadToIntrinsics.cpp (hunk 1 above)
Comment (kuhar) [reply compiler/src/iree/compiler/Preprocessing/Common/PadToIntrinsics.cpp]: Good catch, thanks

**Review by MaheshRavishankar — COMMENTED** [review]
Thanks @kuhar

**Pull request merged**